server/api/resolvers/requestTypeResolvers.ts

```diff
@@ -0,0 +1,188 @@
+                requestGroup.requestTypes.push({ 
+                    _id: newRequestType._id
+                })
+                await requestGroup.save({ session: session })
+
+                return newRequestType
+            })
+        })
+    },
+    updateRequestType: async (_, { requestType }, { authenticateUser }): Promise<RequestTypeInterface> => {
+        return authenticateUser().then(async () => {
+            return sessionHandler(async (session) => {
+                const oldRequestType = await RequestType.findById(requestType._id).session(session)
+                const modifiedRequestTypeObject = new RequestType({...oldRequestType, ...requestType})
+                const newRequestType = await modifiedRequestTypeObject.save({ session: session })
+
+                if (oldRequestType.requestGroup !== newRequestType.requestGroup) {
+                    swapRequestGroupForRequestType(newRequestType, oldRequestType.requestGroup, newRequestType.requestGroup, session)
+                }
+
+                return newRequestType
+            })
+        })
+    },
+    deleteRequestType: async (_, { _id }, { authenticateUser }): Promise<RequestTypeInterface> => {
+        return authenticateUser().then(async () => {
+            const requestType = await RequestType.findById(_id)
+            requestType.deletedAt = new Date()
+            return requestType.save()
+        })
+    },
+    changeRequestGroupForRequestType: async (_, { requestType, requestGroup }, { authenticateUser }): Promise<RequestTypeInterface> => {
+        return authenticateUser().then(async () => {
+            return sessionHandler(async (session) => {
+                const modifiedRequestTypeObject = await RequestType.findById(requestType).session(session)
+
+                if (modifiedRequestTypeObject.requestGroup !== requestGroup) {
+                    swapRequestGroupForRequestType(modifiedRequestTypeObject, requestGroup, modifiedRequestTypeObject.requestGroup, session)
+                }
+
+                modifiedRequestTypeObject.requestGroup = requestGroup
+                return modifiedRequestTypeObject.save({ session: session })
+            })
+        })
+    },
+}
+
+const requestTypeResolvers = {
+    requestGroup: async (parent, __, ___): Promise<RequestGroupInterface> => {
+        return RequestGroup.findById(parent.requestGroup)
+    },
+    requests: async (parent, __, ___): Promise<Array<RequestInterface>> => {
+        return parent.requests.map((requestEmbedding) => {
+            return Request.findById(requestEmbedding._id)
+        })
+    },
+    deleted: (parent, __, ___): boolean => {
+        return parent.deletedAt !== undefined
+    },
+    openRequests: async (parent, __, ___): Promise<Array<RequestInterface>> => {
+        return filterOpenRequestEmbeddings(parent.requests)
+            .map((requestEmbedding) => {
+                return Request.findById(requestEmbedding._id)
```

Comment:
await Request.findById

Reply:
GraphQL awaits this just like the previous comments